codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -369,27 +369,35 @@ protected void doClose() throws Exception {
 
         @Override
         protected void doWrite(Object msg) {
-            if (!(msg instanceof Http2StreamFrame)) {
-                ReferenceCountUtil.release(msg);
-                throw new IllegalArgumentException("Message must be an Http2StreamFrame: " + msg);
-            }
-            Http2StreamFrame frame = (Http2StreamFrame) msg;
-            ChannelPromise promise = ctx.newPromise();
-            if (isStreamIdValid(frame.streamId())) {
-                ReferenceCountUtil.release(frame);
-                throw new IllegalArgumentException("Stream id must not be set on the frame. Was: " + frame.streamId());
-            }
-            if (!isStreamIdValid(streamId())) {
-                if (!(frame instanceof Http2HeadersFrame)) {
-                    throw new IllegalArgumentException("The first frame must be a headers frame. Was: " + frame.name());
+
+            if (msg instanceof Http2StreamFrame) {
+                Http2StreamFrame frame = (Http2StreamFrame) msg;
+                ChannelPromise promise = ctx.newPromise();
+                if (isStreamIdValid(frame.streamId())) {
+                    ReferenceCountUtil.release(frame);
+                    throw new IllegalArgumentException("Stream id must not be set on the frame. Was: "
+                        + frame.streamId());
+                }
+                if (!isStreamIdValid(streamId())) {
+                    if (!(frame instanceof Http2HeadersFrame)) {
+                        throw new IllegalArgumentException("The first frame must be a headers frame. Was: "
```

Comment:
looks like we missed a call to `ReferenceCountUtil.release(frame);` previously ... can you add this?